src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2835,6 +2837,53 @@ public override IOperation VisitCoalesce(ICoalesceOperation operation, int? capt
             return result;
         }
 
+        public override IOperation VisitCoalesceAssignment(ICoalesceAssignmentOperation operation, int? captureIdForResult)
+        {
+            SpillEvalStack();
+
+            RegionBuilder resultCaptureRegion = _currentRegion;
+
+            EvalStackFrame frame = PushStackFrame();
+
+            var whenNull = new BasicBlockBuilder(BasicBlockKind.Block);
+            IOperation testExpression = NullCheckAndConvertCoalesceValue(operation.Target, coalesceConversion: null, operation.Type, whenNull);
+
+            // Coalesce can have conversions on the left side, but coalesce assignment should not.
+            Debug.Assert(testExpression.Kind != OperationKind.Conversion);
```

Comment:
>Debug.Assert(testExpression.Kind != OperationKind.Conversion); [](start = 12, length = 62)

Can ```testExpression``` be a conversion in  an error case?

Reply:
No longer relevant, as I'm not using `NullCheckAndConvertCoalesceValue` anymore.

---
In reply to: [209390571](https://github.com/dotnet/roslyn/pull/29063#discussion_r209390571) [](ancestors = 209390571)